internal/provisioner/objects/rbac/rbac.go

```diff
@@ -53,10 +53,30 @@ func ensureContourRBAC(ctx context.Context, cli client.Client, contour *model.Co
 		return fmt.Errorf("failed to ensure service account %s/%s: %w", contour.Namespace, names.ServiceAccount, err)
 	}
 
-	// Ensure cluster role & binding.
-	if err := clusterrole.EnsureClusterRole(ctx, cli, names.ClusterRole, contour); err != nil {
-		return fmt.Errorf("failed to ensure cluster role %s: %w", names.ClusterRole, err)
+	// By default, Contour watches all namespaces, use default cluster role and rolebinding
+	if contour.WatchAllNamespaces() {
+		// Ensure cluster role & binding.
+		if err := clusterrole.EnsureClusterRole(ctx, cli, names.ClusterRole, contour, false); err != nil {
```

Comment:
Maybe define a variable `gatewayClassOnly` for `false/true`?

Reply:
the `clusterRoleForGatewayclassOnly` is too verbose, the func name `clusterrole.EnsureClusterRole` already contains `ClusterRole`, it can be removed